File: stats.py
from collections import Counter

class StatisticsCalculator:
    def __init__(self, data):
        if not all(isinstance(x, int) for x in data):
            raise ValueError("All elements must be integers")
        self.data = data

    def mean(self):
        return sum(self.data) / len(self.data) if self.data else None

    def median(self):
        if not self.data:
            return None
        sorted_data = sorted(self.data)
        n = len(sorted_data)
        mid = n // 2
        if n % 2 == 1:
            return sorted_data[mid]
        return (sorted_data[mid - 1] + sorted_data[mid]) / 2

    def mode(self):
        if not self.data:
            return []
        freq = Counter(self.data)
        max_count = max(freq.values())
        return [val for val, cnt in freq.items() if cnt == max_count]

def main():
    try:
        raw = input("Enter integers separated by spaces: ").strip()
        data = list(map(int, raw.split()))
    except ValueError:
        print("Invalid input: please enter only integers.")
        return

    calc = StatisticsCalculator(data)
    print(f"\nMean:   {calc.mean():.2f}")
    print(f"Median: {calc.median():.2f}")
    modes = calc.mode()
    print("Mode" + ("s" if len(modes) > 1 else "") + ":", ", ".join(map(str, modes)))

if __name__ == "__main__":
    main()

File: stats.ml
let mean lst =
  let sum = List.fold_left ( + ) 0 lst in
  float_of_int sum /. float_of_int (List.length lst)


let median lst =
  let sorted = List.sort compare lst in
  let n = List.length sorted in
  if n mod 2 = 1 then
    float_of_int (List.nth sorted (n / 2))
  else
    let a = List.nth sorted (n / 2 - 1) in
    let b = List.nth sorted (n / 2) in
    (float_of_int (a + b)) /. 2.0


let mode lst =

  let freq =
    List.fold_left (fun acc x ->
      let count = try List.assoc x acc + 1 with Not_found -> 1 in
      (x, count) :: List.remove_assoc x acc
    ) [] lst
  in

  let max_count =
    List.fold_left (fun m (_, c) -> if c > m then c else m) 0 freq
  in

  freq
  |> List.filter (fun (_, c) -> c = max_count)
  |> List.map fst


let () =
  (* Prompt & read *)
  Printf.printf "Enter integers separated by spaces:\n%!";
  let line = read_line () in
  (* Split on spaces, convert to int *)
  let data = 
    String.split_on_char ' ' line
    |> List.filter ((<>) "")
    |> List.map int_of_string
  in


  let m    = mean data in
  let med  = median data in
  let mods = mode data in


  Printf.printf "\nMean:   %.2f\n" m;
  Printf.printf "Median: %.2f\n" med;
  Printf.printf "Mode%s: %s\n"
    (if List.length mods > 1 then "s" else "")
    (String.concat ", " (List.map string_of_int mods))
